realm/realm-library/src/androidTest/java/io/realm/DynamicRealmObjectTests.java

```diff
@@ -1727,4 +1727,23 @@ public void run() {
         thread.start();
         TestHelper.awaitOrFail(threadFinished);
     }
+
+    @Test
+    public void getNullableBoolean() {
```

Comment:
The fix seems to indicate the bug impacted all primitive types. We should have tests for all of them instead of just boolean.